src/helpers/Sentry.ts

```diff
@@ -50,15 +50,19 @@ export default {
     return typeof(window) !== 'undefined' && config.SENTRY_ENABLED;
   },
 
-  captureException(error: any, level: Sentry.Severity = Severity.Error) {
+  captureException(error: any, level: Sentry.Severity = Severity.Error): string | undefined {
+    let eventId;
```

Comment:
i realized after my suggestion that the types are tighter if you move the `: string | undefined` from the return type to the eventId definition, otherwise i'll be `any`

Reply:
I think it is not because of this
When I've pasted this outside of anything 
```ts
const b = (show: boolean) => {
    let a
    if (show) {
      a = 'asd'
    } else {
      return a
    }
    return a
  }
  ```
  then typescript easily identified that b will return a string or undefined
  
  it seems like there is something wrong with this line:
  ```ts
  eventId = Sentry.captureException(error);
  ```
  When I hover `eventId` here it still says it will be `string | undefined` (as declared after the change) when it should say `string`
  
This also works correctly:
```ts
let eventId;

    if (this.isEnabled) {
      eventId = Sentry.captureException(error);
    } else if (!this.shouldCollectErrors) {
      console.error(error); // tslint:disable-line:no-console
    }

    return eventId;
    ```
    
 so I think it is caused because of wrap in `Sentry.withScope`
    
 Not sure how much do we care, because it seems to be working on the heroku

Reply:
what issue are you talking about? after calling `eventId = Sentry.captureException(error);` it is correct for `eventId` to be `string | undefined`

Reply:
For this code:
```ts
captureException(error: any, level: Sentry.Severity = Severity.Error) {
    let eventId;

    if (this.isEnabled) {
      Sentry.withScope((scope) => {
        scope.setLevel(level);
        eventId = Sentry.captureException(error);
      });
    } else if (!this.shouldCollectErrors) {
      console.error(error); // tslint:disable-line:no-console
    }

    return eventId;
  },
```
typescript is telling that the return type of `captureException` is `undefined`

and for this code:
```ts
captureException(error: any, level: Sentry.Severity = Severity.Error) {
    let eventId;

    if (this.isEnabled) {
      eventId = Sentry.captureException(error);
    } else if (!this.shouldCollectErrors) {
      console.error(error); // tslint:disable-line:no-console
    }

    return eventId;
  },
```
typescript is saying that the return type is `string or undefined`

Reply:
oh yea i think that the addition of the withScope callback breaks that inference, but if you have the definition from the start its ok